### PR TITLE
Add onRest callback

### PR DESCRIPTION
**Pull request by mickaelchanrion**
Hey!
I'm working on a project using tiny-spring and I need to know when a spring animation is completed so I can do things. Would be nice to have that feature. Tell me if I missed something :)

### CODE REVIEW

**Comment by asbjornh** [comment]
Hi! Thanks for the PR! I’m happy to merge this except for one small thing. The name `onComplete` is kind of misleading since a dynamic spring never really “completes”.

I prefer `onRest` as used by for example react-motion. Maybe also update the docs to reflect that (see for instance [this](https://github.com/chenglou/react-motion#--onrest---void)).

If you make those changes I’ll merge and publish a new version tomorrow! :)

File: spring.ts [hunk 1]
```diff
@@ -41,9 +43,14 @@ export default function Spring(
       onUpdate = fn;
       fn(position);
     },
+    onComplete: (fn = v => {}) => {
+      onComplete = fn;
+      fn(position);
```
Comment (asbjornh) [inline spring.ts:48]: Hmm also I’m not sure that this callback should be called immediately after subscription? I would find that a bit confusing but maybe I’m missing something?

**Review by asbjornh — CHANGES_REQUESTED** [review]

**Comment by mickaelchanrion** [comment]
Ok. Good catches! I'm fixing all of that

**Review by asbjornh — APPROVED** [review]

**Comment by asbjornh** [comment]
Cool! Thanks :) I'll publish a new version soon

**Pull request merged**